include/zephyr/drivers/counter.h

```diff
@@ -304,6 +345,92 @@ static inline uint32_t z_impl_counter_get_frequency(const struct device *dev)
 	return api->get_freq ? api->get_freq(dev) : config->freq;
 }
 
+#if defined(CONFIG_COUNTER_CAPTURE) || defined(__DOXYGEN__)
+/**
+ * @brief Set callback function for a counter timestamp capture
+ *
+ * @param dev  Pointer to the device structure for the driver instance
+ * @param chan_id Channel ID
+ * @param flags Configuration flags (COUNTER_CAPTURE_*)
+ * @param cb Callback function reference
+ * @param user_data Argument passed to the callback function
+ *
+ * @retval 0 If successful.
+ * @retval Negative error code on failure
+ */
+__syscall int counter_capture_callback_set(const struct device *dev,
+					   uint8_t chan_id,
+					   uint32_t flags,
+					   counter_capture_cb_t cb,
+					   void *user_data);
+
+static inline int z_impl_counter_capture_callback_set(const struct device *dev,
+						      uint8_t chan_id,
+						      uint32_t flags,
+						      counter_capture_cb_t cb,
+						      void *user_data)
+{
+	const struct counter_driver_api *api =
+				(struct counter_driver_api *)dev->api;
+
+	if (!api->capture_callback_set) {
```

Comment:
```suggestion
	if (api->capture_callback_set == NULL) {
```
Ditto for `z_impl_counter_capture_enable()` and `z_impl_counter_capture_disable()`.

Reply:
Well, the whol file should be updated. Zephyr coding guideline explicitly requires comparison against NULL here.

Reply:
done